package Protobuf;

message MonarchHeader
{
  required string Filename = 1;
  required string Timestamp = 2 [default = "(unknown)"];
  required string Description = 3 [default = "(unknown)"];
  
  enum ContentModes
  {
    Signal = 0;
    Background = 1;
    Other = 2;
  }
  required ContentModes ContentMode = 4 [default = Signal];
  
  enum SourceModes
  {
    Mantis = 0;
    Simulation = 1;
  }
  required SourceModes SourceMode = 5 [default = Mantis];
  
  enum FormatModes
  {
    Single = 0;
    SeparateDual = 1;
    InterleavedDual = 2;
  }  
  required FormatModes FormatMode = 6; 
  
  required double Rate = 7;
  required uint32 Duration = 8;
  required uint32 Length = 9;
}
